forgerock-authenticator/src/main/java/org/forgerock/android/auth/PushNotification.java

```diff
@@ -221,16 +307,79 @@ String serialize() {
     }
 
     /**
-     * Accepts the push authentication request.
-     * @param listener Listener for receiving the HTTP call response code.
+     * Accepts the push authentication request. Use this method to approve notification of
+     * type {@code PushType.DEFAULT}.
+     * @param listener Listener for receiving the authentication result.
      */
     public final void accept(@NonNull FRAListener<Void> listener) {
-        Logger.debug(TAG, "Accept Push Authentication request for message: %s", getMessageId());
-        performAcceptDenyAsync(true, listener);
+        if (this.pushType == PushType.DEFAULT) {
+            Logger.debug(TAG, "Accept Push Authentication request for message: %s", getMessageId());
+            performAcceptDenyAsync(true, listener);
+        } else {
+            listener.onException(new PushMechanismException("Error processing the Push " +
+                    "Authentication request. This method cannot be used to process notification " +
+                    "of type: " + this.pushType));
+        }
+    }
+
+    /**
+     * Accepts the push notification request with the challenge response. Use this method to handle
+     * notification of type {@code PushType.BIOMETRIC}.
+     * @param challengeResponse the response for the Push Challenge
+     * @param listener Listener for receiving the authentication result.
+     */
+    public final void accept(@NonNull String challengeResponse, @NonNull FRAListener<Void> listener) {
+        if (this.pushType == PushType.CHALLENGE) {
+            Logger.debug(TAG, "Respond the challenge for message: %s", getMessageId());
+            PushResponder.getInstance().authenticationWithChallenge(this, challengeResponse, listener);
+        } else {
+            listener.onException(new PushMechanismException("Error processing the Push " +
+                    "Authentication request. This method cannot be used to process notification " +
+                    "of type: " + this.pushType));
+        }
     }
 
     /**
-     * Deny the push authentication request.
+     * Accepts the push notification request with Biometric Authentication. Use this method to handle
+     * notification of type {@code PushType.BIOMETRIC}.
+     * @param title the title to be displayed on the prompt.
+     * @param subtitle the subtitle to be displayed on the prompt.
+     * @param allowDeviceCredentials if {@code true}, accepts device PIN, pattern, or password to process notification.
+     * @param activity the activity of the client application that will host the prompt.
+     * @param listener listener for receiving the push authentication result.
+     */
+    @RequiresApi(Build.VERSION_CODES.M)
```

Comment:
i think we should talk about bumping out min SDK to 23 ? .

Reply:
it's a good idea